datafusion/physical-expr/src/partitioning.rs

```diff
@@ -331,12 +417,161 @@ mod tests {
-                Distribution::HashPartitioned(_) => {
+                Distribution::HashPartitioned(_) | Distribution::KeyPartitioned(_) => {
                     assert_eq!(result, (true, false, false, true, false))
                 }
             }
         }
 
         Ok(())
     }
+
+    #[test]
+    fn test_key_partitioned_satisfies() {
+        use crate::expressions::Column;
+        use std::sync::Arc;
+
+        let schema = Arc::new(Schema::new(vec![
+            Field::new("a", DataType::Int32, false),
+            Field::new("b", DataType::Int32, false),
+            Field::new("c", DataType::Int32, false),
+        ]));
+
+        let col_a = Arc::new(Column::new("a", 0)) as Arc<dyn PhysicalExpr>;
+        let col_b = Arc::new(Column::new("b", 1)) as Arc<dyn PhysicalExpr>;
+        let col_c = Arc::new(Column::new("c", 2)) as Arc<dyn PhysicalExpr>;
+
+        let eq_properties = EquivalenceProperties::new(schema);
+
+        // Case 1: Exact match - [a] satisfies [a]
+        let partitioning = Partitioning::KeyPartitioned(vec![Arc::clone(&col_a)], 4);
+        let required = vec![Arc::clone(&col_a)];
+        assert!(partitioning
+            .satisfy(&Distribution::HashPartitioned(required), &eq_properties));
+
+        // Case 2: Superset - [a] satisfies [a, b] (partition columns are subset of required)
+        let partitioning = Partitioning::KeyPartitioned(vec![Arc::clone(&col_a)], 4);
+        let required = vec![Arc::clone(&col_a), Arc::clone(&col_b)];
+        assert!(partitioning
+            .satisfy(&Distribution::HashPartitioned(required), &eq_properties));
+
+        // Case 3: NOT satisfied - [a, b] does not satisfy [a]
+        let partitioning =
+            Partitioning::KeyPartitioned(vec![Arc::clone(&col_a), Arc::clone(&col_b)], 4);
+        let required = vec![Arc::clone(&col_a)];
+        assert!(!partitioning
+            .satisfy(&Distribution::HashPartitioned(required), &eq_properties));
+
+        // Case 4: Multi-column exact match
+        let partitioning =
+            Partitioning::KeyPartitioned(vec![Arc::clone(&col_a), Arc::clone(&col_b)], 4);
+        let required = vec![Arc::clone(&col_a), Arc::clone(&col_b)];
+        assert!(partitioning
+            .satisfy(&Distribution::HashPartitioned(required), &eq_properties));
+
+        // Case 5: Multi-column superset
+        let partitioning =
+            Partitioning::KeyPartitioned(vec![Arc::clone(&col_a), Arc::clone(&col_b)], 4);
+        let required = vec![Arc::clone(&col_a), Arc::clone(&col_b), Arc::clone(&col_c)];
+        assert!(partitioning
+            .satisfy(&Distribution::HashPartitioned(required), &eq_properties));
+
+        // Case 6: Different columns - not satisfied
+        let partitioning = Partitioning::KeyPartitioned(vec![Arc::clone(&col_a)], 4);
+        let required = vec![Arc::clone(&col_b)];
+        assert!(!partitioning
+            .satisfy(&Distribution::HashPartitioned(required), &eq_properties));
+    }
```

Comment:
Can you add these edge test cases:
1. partitioning is empty, required is not
2. partitioning is not empty, required is
3. both empty